### DISCUSSION TITLE
Isotype identification

### DISCUSSION THREAD

**Discussion by ligalaizik** [milaboratory/mixcr#1038] [Q&A]
Hi, when using the mixcr align command and exportAlignments, some of the sequence's isotypes are not identified. In aligning those sequences manually with IMGT, I found that some unidentified sequences are IgGs, but the sub-isotype cannot be identified. So instead of assigning them as IGHG, mixcr didn't assign them any isotypes. Can that be fixed? So sequences that are Iggs, but don't have enough information to determine their sub-isotype, will just be assigned as IgGs?

**Comment by mizraelson** [comment]
Hi, can you please share the exact commands you use ( `mixcr align`  and `mixcr exportAlignments`). Also if its possible can you please share an example of such read sequence.

**Comment by ligalaizik** [comment]
Command :

mixcr align -s mmu -r report_232.txt M1-D1-D21-232_S3_L001_R1_001.fastq.gz M1-D1-D21-232_S3_L001_R2_001.fastq.gz output_align_232.vdjca --verbose -f

mixcr exportAlignments -f -nFeature {FR1Begin:FR4End} -targets -vHit -dHit -jHit -cHit -vAlignment -dAlignment -jAlignment -cAlignments -nFeature FR1 -nFeature CDR1 -nFeature FR2 -nFeature CDR2 -nFeature FR3 -nFeature CDR3 -nFeature FR4    -aaFeature FR1 -aaFeature CDR1 -aaFeature FR2 -aaFeature CDR2 -aaFeature FR3 -aaFeature CDR3 -aaFeature FR4    output_align_232.vdjca  output_align_232.txt 

Example of read sequence :
R1:
ACCTCCTTTAATTCCCGAGGTGCAGCTCCAGCAGTCTGGGACTGTGCTGGCAAGGCCTGGGGCTTCCGTGAAGATGTCCTGCAAGGCTTCTGGCTACAGCTTTACCAGCTACTGGATGCACTGGGTAAAACAGAGGCCTGGACAGGGTCTAGAATGGATTGGTGCTATTTATCCTGGAAATAGTGATACTAGCTACAACCAGAAGTTCAAGGGCAAGGCCAAACTGACTGCAGTCACATCCGCCAGCACTGCCTACATGGAGCTCAGCAGCCTGACAAATGAGGACTCTGCGGTCTATTAC
R2:
ACCGGAGGAGAGAGAGAGAGACATCGCCAAGTGGATAGACCGATGGGGGTGTCGTTTTGGCTGAGGAGACTGTGAGAGTGGTGCCTTGGCCCCAGTAGTCACGTCGTAACCAGGTACAGTAATAGACCGCAGAGTCCTCATTTGTCAGGCTGCTGAGCTCCATGTAGGCAGTGCTGGCGGATGTGACTGCAGTCAGTTTGGCCTTGCCCTTGAACTTCTGGTTGTAGCTAGTATCACTATTTCCAGGATAAATAGCACCAATCCATTCTAGACCCTGTCCGGGCCGCTGTATTACCCAGTG

Thanks !

**Comment by mizraelson** [comment]
It seems you are using an old mixcr version. I suggest using the latest available version.

Also, there is an unaligned sequence (barcode?) of 29 nucleotides at the end of R2 read which interferes the global alignment. In the command bellow I use the parameter `--rigid-right-alignment-boundary C` which turns on semi-local alignment for the C gene and it works:
 
```
mixcr align \
    --preset generic-bcr-amplicon \
    --species mmu \
    --floating-left-alignment-boundary \
    --rigid-right-alignment-boundary C \
    --rna \
    input_R1.fastq.gz input_R2.fastq.gz  \
    result.vdjca
```

Then your `mixcr exportAlignments` command properly returns the isotype:

<img width="410" alt="image" src="https://user-images.githubusercontent.com/18702359/224749828-2a7a0fa2-211b-4c81-a606-2de897752009.png">

**Comment by ligalaizik** [comment]
I prefer using the old version because the new version’s output (after export alignments) is different, and all my downstream pipeline depends on the older version output.Is there a way to use this command with my oldest version ( v4.0.0b)? So I would get the exact same output but the c hit will be more accurate? Thanks! Ligal. ‫ב-13 במרץ 2023, בשעה 17:35, ‏‏mizraelson ***@***.***> כתב/ה:‬﻿
It seems you are using an old mixcr version. I suggest using the latest available version.
Also, there is an unaligned sequence (barcode?) of 29 nucleotides at the end of R2 read which interferes the global alignment. In the command bellow I use the parameter --rigid-right-alignment-boundary C which turns on local alignment for the C gene and it works:
mixcr align \
    --preset generic-bcr-amplicon \
    --species mmu \
    --floating-left-alignment-boundary \
    --rigid-right-alignment-boundary C \
    --rna \
    input_R1.fastq.gz input_R2.fastq.gz  \
    result.vdjca

Then your mixcr exportAlignments command properly returns the isotype:

**Comment by mizraelson** [comment]
In the old mixcr version you can try:

```
mixcr align \
    -OjParameters.parameters.floatingRightBound=false \
    -OcParameters.parameters.floatingLeftBound=true \
    .
    .
    .
```

Or, for `mixcr analyze amplicon`:

` --3-end c-primers`
